src/back/hlsl/writer.rs

```diff
@@ -1859,12 +1816,10 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
                 let func_name = &self.names[&NameKey::Function(function)];
                 write!(self.out, "{func_name}(")?;
                 for (index, argument) in arguments.iter().enumerate() {
-                    self.write_expr(module, *argument, func_ctx)?;
-                    // Only write a comma if isn't the last element
-                    if index != arguments.len().saturating_sub(1) {
-                        // The leading space is for readability only
+                    if index != 0 {
                         write!(self.out, ", ")?;
                     }
+                    self.write_expr(module, *argument, func_ctx)?;
                 }
```

Comment:
I'm not a fan of `write_slice` since it takes a closure. Maybe we could instead introduce a new type that implements Iterator (and wraps around other Iterators) that writes the comma after each iteration.

Reply:
In chat we agreed no change needed here.